pandas/core/arrays/sparse/array.py

```diff
@@ -830,6 +831,12 @@ def _first_fill_value_loc(self):
         diff = np.r_[np.diff(indices), 2]
         return indices[(diff > 1).argmax()] + 1
 
+    @doc(ExtensionArray.duplicated)
+    def duplicated(
+        self, keep: Literal["first", "last", False] = "first"
+    ) -> npt.NDArray[np.bool_]:
+        return algos.duplicated(np.asarray(self), keep=keep)
```

Comment:
Does sparse need to pass mask for it's NA values?

Reply:
updated to pass the mask